test/Sentry.Tests/TaskUnobservedTaskExceptionIntegrationTests.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using NSubstitute;
+using Sentry.Integrations;
+using Sentry.Internal;
+using Xunit;
+using System.Linq;
+using System.Threading.Tasks;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+
+namespace Sentry.Tests
+{
+    public class TaskUnobservedTaskExceptionIntegrationTests
+    {
+        private class Fixture
+        {
+            public IHub Hub { get; set; } = Substitute.For<IHub, IDisposable>();
+            public IAppDomain AppDomain { get; set; } = Substitute.For<IAppDomain>();
+
+            public Fixture() => Hub.IsEnabled.Returns(true);
+
+            public TaskUnobservedTaskExceptionIntegration GetSut()
+                => new TaskUnobservedTaskExceptionIntegration(AppDomain);
+        }
+
+        private readonly Fixture _fixture = new Fixture();
+        public SentryOptions SentryOptions { get; set; } = new SentryOptions();
+
+        [Fact]
+        public void Handle_WithException_CaptureEvent()
+        {
+            var sut = _fixture.GetSut();
+            sut.Register(_fixture.Hub, SentryOptions);
+
+            sut.Handle(this, new UnobservedTaskExceptionEventArgs(new AggregateException()));
+
+            _ = _fixture.Hub.Received(1).CaptureEvent(Arg.Any<SentryEvent>());
+        }
+
```

Comment:
Would be nice to have "an integration test" like actually throwing a unobserved task exception and seeing if it works.

I wrote some code (on GitHub so take it as a pseudo-code) to show the idea.
Would something like that make sense?

```suggestion
        [Fact]
        public void Handle_WithException_CaptureEvent()
        {
            var sut = _fixture.GetSut();
            sut.Register(_fixture.Hub, SentryOptions);

            sut.Handle(this, new UnobservedTaskExceptionEventArgs(new AggregateException()));

            _ = _fixture.Hub.Received(1).CaptureEvent(Arg.Any<SentryEvent>());
        }

        [Fact] // Integration test
        public void Handle_UnobservedTaskException_CaptureEvent()
        {
            _fixture.AppDomain = AppDomainAdapter.Instance;
             var evt = new ManualResetEvent();
            _fixture.Hub.CaptureEvent(Arg.Any<SentryEvent>()).Callback(() => evt.Set());
            var sut = _fixture.GetSut();
            sut.Register(_fixture.Hub, SentryOptions);
            try 
            {
                Task.Run(() => throw null);

                Assert.True(evt.Wait(TimeSpan.FromMilliseconds(100));
            }
            finally 
            {
                sut.Unregister(_fixture.Hub);
            }
        }

```